src/game/Game.java

```diff
@@ -357,59 +353,55 @@ public void printResults() {
                     // AND NOW FOR THE DIAGONALS //
 
                     // down + right  ++
-                    if (i > 0) {
+                    if (i < 6 && j < 6) {
                         while (board[row + 1][col + 1] == opColor && row < 6 && col < 6) {
                             row++;
                             col++;
                         }
-                        if (board[row + 1][col + 1] == EMPTY) {
-                            move[0] = row + 1;
-                            move[1] = col + 1;
+                        if (board[row + 1][col + 1] == EMPTY && board[row][col] == opColor) {
+                            moves.add(new int[]{row + 1, col + 1});
                         }
                         //reset row and col
                         row = i;
                         col = j;
                     }
 
                     // down + left  +-
-                    if (i > 0) {
-                        while (board[row + 1][col - 1] == opColor && row < 6 && col > 0) {
+                    if (i < 6 && j > 1) {
+                        while (board[row + 1][col - 1] == opColor && row < 6 && col > 1) {
```

Comment:
Need one space to place our piece.
(0 is out of bounds on the column in next if check)